### DISCUSSION TITLE
future of this library

### DISCUSSION THREAD

**Discussion by alexted** [adriangb/di#118] [General]
[dependency-injector](https://github.com/ets-labs/python-dependency-injector) is dead. don't you think you should start promoting your library more actively in the community? For example, writing articles on sites like reddit, dev.to, medium and others, mentioning your library more often in groups, channels, and community threads, speaking at conferences, and so on.

**Comment by adriangb** [comment]
That's a possibility! It's all a balance of time and energy. Currently, I'm very busy working on Pydantic / Pydantic the company so I'm relatively saturated with open source. And no one is sponsoring this project, which to me is the real indicator of a need and adoption.

What I would like for this library is some refactoring and cleanup, solving of the open issues, and some co-maintainers. With that I think it could go 1.0 and promoting it would be much more interesting. I realize that giving talks and writing blog posts it is one way to get there, but I'm also happy to see where organic growth goes given that the constraint is mostly my time right now.

That said I'd be happy to see anyone else give talks, make PRs, etc. and could even review some of it. Maybe you can give it a shot if you like the library and have the time to allocate 😄 ?

**Reply by adriangb** [reply]
Also why do you say that `dependency-injector` is dead?

**Reply by alexted** [reply]
look at the number of open unanswered issues, the date of the last commit and if you are not convinced, read these issues:
https://github.com/ets-labs/python-dependency-injector/issues/688, 
https://github.com/ets-labs/python-dependency-injector/issues/742, 
https://github.com/ets-labs/python-dependency-injector/issues/763

**Reply by adriangb** [reply]
I'm not going to dive in-depth and I'm not going to make declarations on behalf of another project but the maintainer seems to be saying it's not dead, I'd like to be charitable towards them.

In any case, I do think it's fair to say that the library has a number of issues and missing features. The real question is if this library is really any better or if it's just trading off those bugs / missing features for others that just haven't surfaced yet. We can't know!

**Comment by alexted** [comment]
> With that I think it could go 1.0 and promoting it would be much more interesting.

Yes, publishing v1 and removing the notification that the library is still in development is a good idea - it will greatly reduce the level of doubt of people who are familiarizing themselves with it.

**Reply by adriangb** [reply]
Yep I'd love to do that but like I said that's dependent on:
- Co maintainers. I want to be able to disappear off the face of the earth and have any 1.0 project I make live on. A bus factor of 1 doesn't work if I get hit by the bus.
- Fixing of open issues. There are some that I think may require API changes.
- Refactoring and simplifying. Some things in this library are certainly too complex and changing them likely means breaking changes.
- More documentation
- Maybe some level of sponsorship to ensure sustainable development
